src/Template/Loggers/IOHprofiler_ecdf_logger.hpp

```diff
@@ -191,12 +191,44 @@ const IOHprofiler_AttainSuite& IOHprofiler_ecdf_logger<T>::data()
 }
 
 template<class T>
-const IOHprofiler_AttainMat& IOHprofiler_ecdf_logger<T>::at(size_t problem_id, size_t instance_id, size_t dim_id) const
+const IOHprofiler_AttainMat& IOHprofiler_ecdf_logger<T>::at(size_t problem_id, size_t instance_id, size_t dim_id, size_t run_id) const
 {
     assert(_ecdf_suite.count(problem_id) != 0);
     assert(_ecdf_suite.at(problem_id).count(dim_id) != 0);
     assert(_ecdf_suite.at(problem_id).at(dim_id).count(instance_id) != 0);
-    return _ecdf_suite.at(problem_id).at(dim_id).at(instance_id);
+    assert(_ecdf_suite.at(problem_id).at(dim_id).at(instance_id).count(run_id) != 0);
+    return _ecdf_suite.at(problem_id).at(dim_id).at(instance_id).at(run_id);
+}
+
+template<class T>
+IOHprofiler_AttainMat IOHprofiler_ecdf_logger<T>::at(size_t problem_id, size_t instance_id, size_t dim_id) const
+{
+    size_t runs;
+    IOHprofiler_AttainMat mat = _empty;
+    std::vector< std::vector< int >> c_mat(_range_error.size(),std::vector<int>( _range_evals.size(),0));
```

Comment:
`int` should be `size_t`, because it is a counter, it is not supposed to have negative values. `size_t` is a good proxy to the positive integer with the largest span.